fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/ReplaceColumnRefRewriter.java

```diff
@@ -56,7 +56,11 @@ public ScalarOperator visitVariableReference(ColumnRefOperator column, Void cont
             ScalarOperator mapperOperator = operatorMap.get(column).clone();
             if (isRecursively) {
                 while (mapperOperator.getChildren().isEmpty() && operatorMap.containsKey(mapperOperator)) {
-                    mapperOperator = operatorMap.get(mapperOperator).clone();
+                    ScalarOperator mapped = operatorMap.get(mapperOperator);
+                    if (mapped.equals(mapperOperator)) {
```

Comment:
add test cases for this bug.

Reply:
I have added this. I found this bug because I add a new UT

Reply:
your bug because the operatorMap like this?
```
column1 : column2
column2: column2
```

it‘s not a valid input, maybe we need found reason?